packages/x-date-pickers/src/internals/hooks/useField/useField.types.ts

```diff
@@ -59,6 +59,11 @@ export interface UseFieldInternalProps<TValue, TError> {
    * @default false
    */
   readOnly?: boolean;
+  /**
+   * If `true`, the field is disabled.
+   * @default false
+   */
+  disabled?: boolean;
```

Comment:
Right now `useField` is not using `disabled`

Maybe I'm missing something, but I think we should either;

1. Handle `disabled` in `useField` (add it to `UseFieldInternalProps` and return it in `UseFieldResponse`), which should allow to remove it from the `forwardedProps` and avoid the `other.disabled` you added.

2. Handle `disabled` in the components

Is there something preventing us from doing `2.` ? Given that `disabled` is at the root level of the `TextField`, shouldn't it behave like any other prop `useField` do not use ? (`variant` for example)

Reply:
> Handle disabled in the components

Isn't it already handled in components? For single input and base (Date, Time, and DateTime) fields this prop is transitively passed down.
Did you mean to remove it from `useField` and put it somewhere in `MultiInput` fields? As they are the ones who need this prop explicitly defined. 🤔

Reply:
But if is handled by the component, why do we need to define its type here ?

Reply:
> Did you mean to remove it from useField and put it somewhere in MultiInput fields? As they are the ones who need this prop explicitly defined. 

Yes
To avoid having the prop defined in an interface of `useField` when `useField` never care about it

Reply:
> Unless we want to define it here to have a better JSDoc

Well... I was also debating on the JSDoc for it... Changing from __If `true`, the component is disabled.__ to __If `true`, the field is disabled.__ makes sense for Field components or single input Range fields, but for multi input ones it is a bit misleading... :/ 
It would probably make more sense staying with the original `TextFieldProps.disabled` JSDoc 🙈

Reply:
[Moved the disabled type declaration](https://github.com/mui/mui-x/pull/7135/commits/9d2f3f42610e0bd266b49eef08bc80748697c761) to be explicit only for `range` fields.
Seems to simplify the code quite a lot.
Thanks for very insightful comments! 🙏